Make GraalVM Java 11 the default image

gsmet opened the pull request.

gsmet committed:
Make GraalVM Java 11 the default image

gsmet commented:
@gwenneg I let you have a look and merge if it looks OK to you.

gwenneg commented:
Ok @gsmet, review in progress.

gwenneg commented on docs/src/main/asciidoc/native-and-ssl.adoc:
```diff
@@ -206,7 +206,7 @@ You can for example modify your `Dockerfile.native` as follows to copy the requi
 [source, subs=attributes+]
 ----
-FROM quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java8 as nativebuilder
+FROM quay.io/quarkus/ubi-quarkus-native-image:{graalvm-flavor} as nativebuilder
```
This doc section is about Java 8 but the default `{graalvm-flavor}` is `...-java11`. Is that OK to keep things that way?

gwenneg commented:
Should we mention `{graalvm-flavor}` in [README.adoc](https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/README.adoc)?

gwenneg commented on docs/src/main/asciidoc/building-native-image.adoc:
```diff
@@ -369,7 +369,7 @@ Save this file in `src/main/docker/Dockerfile.multistage` as it is not included
 [WARNING]
 ====
-Replace the -java8 in the base image with -java11 if you want to use Java 11.
+Replace `-java11` in the base image with `-java8` if you want to use Java 8.
```
This doesn't really make sense now that `-java8` is gone (and `-java11` does not exist AFAICS).

gsmet replied:
I'm removing that mention.

gwenneg commented on docs/src/main/asciidoc/deploying-to-openshift-s2i.adoc:
```diff
@@ -52,7 +52,7 @@ curl $URL/hello/greeting/quarkus
 ====
 The `oc new-app` command above uses a builder image compatible with JDK 8.
-In order to create a JDK 11 native compatible image use `quay.io/quarkus/ubi-quarkus-native-s2i:{graalvm-version}-java11` as a builder image.
+In order to create a JDK 11 native compatible image use `quay.io/quarkus/ubi-quarkus-native-s2i:{graalvm-flavor}` as a builder image.
```
This one doesn't feel right too.

gsmet replied:
Right let's drop it. Search and replace was a bit too extreme apparently :).

gwenneg commented on docs/src/main/asciidoc/deploying-to-openshift-s2i.adoc:
```diff
@@ -117,11 +117,11 @@ The end result is an image that is less than 40 MB in size (compressed) and does
 ----
 
-Like in previous commands, use `quay.io/quarkus/ubi-quarkus-native-s2i:{graalvm-version}-java11` for building a JDK 11 compatible image.
+Like in previous commands, use `quay.io/quarkus/ubi-quarkus-native-s2i:{graalvm-flavor}` for building a JDK 11 compatible image.
```
Same here.

gsmet replied:
Dropped.

gwenneg commented on docs/src/main/asciidoc/maven-tooling.adoc:
```diff
@@ -371,8 +371,8 @@ Note that in this case the build itself runs in a Docker container too, so you d
-By default, the native image will be generated using the `quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java8` Docker image which is only compatible with JDK 8.
-If you want to build a native image compatible with JDK 11, you will have to use the `-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java11` build argument.
+By default, the native image will be generated using the `quay.io/quarkus/ubi-quarkus-native-image:{graalvm-flavor}` Docker image which is only compatible with JDK 8.
+If you want to build a native image compatible with JDK 11, you will have to use the `-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:{graalvm-flavor}` build argument.
```
The image URL will be the same for both JDK here, there's no reason to keep that tip in its current state (but maybe it could be rewritten differently).

gsmet replied:
Same here.

gwenneg commented on docs/src/main/asciidoc/gradle-tooling.adoc:
```diff
@@ -283,8 +283,8 @@ Note that in this case the build itself runs in a Docker container too, so you d
-By default, the native image will be generated using the `quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java8` Docker image which is only compatible with JDK 8.
-If you want to build a native image compatible with JDK 11, you will have to use the `-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java11` build argument.
+By default, the native image will be generated using the `quay.io/quarkus/ubi-quarkus-native-image:{graalvm-flavor}` Docker image which is only compatible with JDK 8.
+If you want to build a native image compatible with JDK 11, you will have to use the `-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:{graalvm-flavor}` build argument.
```
The image URL will be the same for both JDK here, there's no reason to keep that tip in its current state (but maybe it could be rewritten differently).

gsmet replied:
I dropped it. We want to deprecate Java 8 so let's move everyone to JDK 11.

gsmet replied:
Hum. I thought a bit more about it and came up with another version of it.

gsmet commented:
@gwenneg sorry about the mess! Can you have another look. I adjusted a few things to be more future proof.

Also I removed a mention of the SunEC library as we think we don't need it anymore, right?

gsmet committed:
Report the changes made in Dockerfile.native in the SSL doc

gwenneg commented on docs/src/main/asciidoc/README.adoc:
```diff
@@ -44,5 +44,6 @@ complete list of externalized variables for use is given in the following table:
 
-|\{graalvm-version}|{graalvm-version}| Recommended Graal VM version to use.
+|\{graalvm-version}|{graalvm-version}| Recommended GraalVM version to use.
+|\{graalvm-flavor}|{graalvm-flavor}| The full flavor of GraaVM to use e.g. `19.3.1-java11`.
```
This is unrelated to this PR, but why do we need two columns (`Property Name` and `Value`) while they contain the exact same content for all lines of the table?

gsmet replied:
They do not. One is escaped, not the other.

gwenneg replied:
Oh I missed that detail, sorry :)

gwenneg replied:
I can see the HTML with different values in `quarkus-documentation-999-SNAPSHOT.zip`, but is this file actually published somewhere on a public website? I can't find it anywhere on `https://quarkus.io/`.

gwenneg replied:
On GH, there isn't any difference in terms of display betweens columns 1 and 2.

gsmet replied:
No, it's not. It's just for reference, we don't publish it.

That would be a GH bug :).

gwenneg replied:
Then I still don't understand why we need the `Value` column.

gwenneg commented on docs/src/main/asciidoc/gradle-tooling.adoc:
```diff
@@ -283,8 +283,12 @@ Note that in this case the build itself runs in a Docker container too, so you d
+If you want to build a native executable with a different Docker image (for instance to use a different GraalVM version),
+use the `-Dquarkus.native.builder-image=<image name>` build argument.
+
+The list of the available Docker images can be found on https://quay.io/repository/quarkus/ubi-quarkus-native-image?tab=tags[quay.io]. Note that not all the images are compatible with a given Quarkus version.
```
The last part is not easy to read, I would change it to something like:

```suggestion
The list of the available Docker images can be found on https://quay.io/repository/quarkus/ubi-quarkus-native-image?tab=tags[quay.io]. Note that a given Quarkus version is not compatible with all the images.
```

gwenneg commented on docs/src/main/asciidoc/maven-tooling.adoc:
```diff
@@ -371,8 +371,12 @@ Note that in this case the build itself runs in a Docker container too, so you d
+If you want to build a native executable with a different Docker image (for instance to use a different GraalVM version),
+use the `-Dquarkus.native.builder-image=<image name>` build argument.
+
+The list of the available Docker images can be found on https://quay.io/repository/quarkus/ubi-quarkus-native-image?tab=tags[quay.io]. Note that not all the images are compatible with a given Quarkus version.
```
```suggestion
The list of the available Docker images can be found on https://quay.io/repository/quarkus/ubi-quarkus-native-image?tab=tags[quay.io]. Note that a given Quarkus version is not compatible with all the images.
```

gwenneg commented on docs/src/main/asciidoc/native-and-ssl.adoc:
```diff
@@ -199,42 +199,31 @@ Don't forget to move your GraalVM directory back to where it was.
-FROM quay.io/quarkus/ubi-quarkus-native-image:{graalvm-version}-java11 as nativebuilder
-RUN mkdir -p /tmp/ssl-libs/lib \
-  && cp /opt/graalvm/lib/security/cacerts /tmp/ssl-libs \
-  && cp /opt/graalvm/lib/libsunec.so /tmp/ssl-libs/lib/
```
I don't know if there was a mistake in the previous version, but the path where the certs were copied was different.

gsmet replied:
I decided to change the name as it doesn't contain any library anymore.

gsmet commented:
@gwenneg I took another shot.

gwenneg commented:
> Also I removed a mention of the SunEC library as we think we don't need it anymore, right?

That's correct.

gwenneg reviewed (approved):
LGTM

gsmet commented:
@gwenneg thanks for the thorough review!
